src/Overlay.tsx

```diff
@@ -128,7 +155,15 @@ const Overlay = React.forwardRef<HTMLDivElement, OverlayProps>(
           height={height}
           {...rest}
           ref={combinedRef}
-          visibility={visibility}
+          style={
+            {
+              '--slide-animation-x-factor': slideAnimationFactors.x,
+              '--slide-animation-y-factor': slideAnimationFactors.y,
+              '--slide-animation-timing':
+                slideAnimationFactors.x === 0 && slideAnimationFactors.y === 0 ? '0' : undefined // 0 if we don't want to animate, fallback in css if we do
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            } as any
```

Comment:
```suggestion
            } as React.CSSProperties
```

Reply:
Nice!  It didn't like the unknown keys without a type cast, but it appears to handle it fine if I explicitly cast it as you suggested.  This is specifically called out in the react docs, but they didn't have a suggested approach like this one (tho I think they should): https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d0cd19269f13cffe248bc1439ad6ae959d70524c/types/react/index.d.ts#L1543-L1552 ->  https://github.com/frenic/csstype#what-should-i-do-when-i-get-type-errors